### PR TITLE
Implementation of the App | Алексеев Артур

### PR DESCRIPTION
**Pull request by arthur100500**

### CODE REVIEW

File: src/reqs.txt [hunk 1]
```diff
@@ -0,0 +1,71 @@
+altair==4.2.0
```
Comment (SpirinEgor) [inline src/reqs.txt]: General advice is to keep here only project-specific requirements, e.g. packages that you are importing.
In this way, someone can easily see what technologies you use in the repo and install them manually. Also, your environment may contain packages that are already unrelated to the app (move to other libs), dev packages (black, mypy), or hardware-specific (cudnn-c11).

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    sidebar = st.sidebar
+    sidebar.header("Configure cockatoo")
+    sidebar.subheader("Cockatoo parameters")
+
+    parrot.set_seed(int(sidebar.number_input("Seed", 0, 999999, 1)))
+    parrot.adequacy_threshold = sidebar.slider("Adequacy", 0.0, 1.0, 0.8)
+    parrot.fluency_threshold = sidebar.slider("Fluency", 0.0, 1.0, 0.8)
+    parrot.amount = int(sidebar.slider("Maximum amount", 1, 9, 3))
+    parrot.do_diverse = sidebar.checkbox("Do Diverse?")
```
Comment (SpirinEgor) [inline src/main.py]: That doesn't look like a good way to interact between objects. To use class someone need to separately change values (moreover, there are no getters/setters to ensure invariance) and then call the method to generate.

I suggest you pass these values in `para_phrase` method together with text input.

**Review by SpirinEgor — CHANGES_REQUESTED** [review]
Good solution, minimum comments 👍 Let's make interfaces more strict for clean and easy usage.

Also, I see that you are using CI. It is bad that you added it without PR because I can't validate the correctness. You are using some solution from the marketplace, which I don't know. I suggest you update it and replace it with straightforward usage of black. Why do you need a whole external dependency to replace 2 lines in YAML? 🙃

File: src/main.py (hunk 2 above)
Comment (arthur100500) [reply src/main.py]: Done. Now para_phrase takes 4 more positional arguments, which were implemented as a fields in `ParrotModel`

File: src/reqs.txt (hunk 1 above)
Comment (arthur100500) [reply src/reqs.txt]: I added all the dependencies just after creating venv and importing libs that were necessary. Seems like `altair` is part of `streamlit` module, and if I remove it it refuses to start

Comment (arthur100500) [reply src/reqs.txt]: I removed everything except for `parrot` from git and `streamlit`. Works with only these two 👍

**Review by SpirinEgor — APPROVED** [review]
Nice application 👍

**Pull request merged**